velox/substrait/tests/VeloxSubstraitRoundTripPlanConverterTest.cpp

```diff
@@ -0,0 +1,106 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <folly/Random.h>
+
+#include "velox/exec/tests/utils/OperatorTestBase.h"
+#include "velox/exec/tests/utils/PlanBuilder.h"
+#include "velox/vector/tests/VectorMaker.h"
+
+#include "velox/substrait/SubstraitToVeloxPlan.h"
+#include "velox/substrait/VeloxToSubstraitPlan.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::exec;
+using namespace facebook::velox::exec::test;
+using namespace facebook::velox::substrait;
+
+class VeloxSubstraitRoundTripPlanConverterTest : public OperatorTestBase {
+ protected:
+  /// Makes a vector of INTEGER type with 'size' RowVectorPtr.
+  /// @param size The number of RowVectorPtr.
+  /// @param childSize The number of columns for each row.
+  /// @param batchSize The batch Size of the data.
+  std::vector<RowVectorPtr>
+  makeVectors(int64_t size, int64_t childSize, int64_t batchSize) {
+    std::vector<RowVectorPtr> vectors;
+    std::mt19937 gen(std::mt19937::default_seed);
+    for (int i = 0; i < size; i++) {
+      std::vector<VectorPtr> children;
+      for (int j = 0; j < childSize; j++) {
+        children.emplace_back(makeFlatVector<int32_t>(
+            batchSize,
+            [&](auto row) {
+              return folly::Random::rand32(INT32_MAX / 4, INT32_MAX / 2, gen);
+            },
+            nullEvery(2)));
+      }
+
+      vectors.push_back(makeRowVector({children}));
+    }
+    return vectors;
+  };
+
+  void assertPlanConversion(
+      const std::shared_ptr<const core::PlanNode>& plan,
+      const std::string& duckDbSql) {
+    assertQuery(plan, duckDbSql);
+    // Convert Velox Plan to Substrait Plan.
+    google::protobuf::Arena arena;
+    auto substraitPlan = veloxConvertor_->toSubstrait(arena, plan);
+
+    // Convert Substrait Plan to Velox Plan.
+    auto veloxPlan =
+        substraitConverter_->toVeloxPlan(substraitPlan, pool_.get());
+
+    // Assert velox again.
+    assertQuery(veloxPlan, duckDbSql);
+  }
+
+  std::shared_ptr<VeloxToSubstraitPlanConvertor> veloxConvertor_ =
+      std::make_shared<VeloxToSubstraitPlanConvertor>();
+  std::shared_ptr<SubstraitVeloxPlanConverter> substraitConverter_ =
+      std::make_shared<SubstraitVeloxPlanConverter>();
+  std::unique_ptr<memory::ScopedMemoryPool> pool_{
+      memory::getDefaultScopedMemoryPool()};
+};
+
+TEST_F(VeloxSubstraitRoundTripPlanConverterTest, project) {
+  auto vectors = makeVectors(3, 4, 2);
+  createDuckDbTable(vectors);
+  auto plan = PlanBuilder().values(vectors).project({"c0", "c1"}).planNode();
```

Comment:
This project is a no-op. If conversion skips project operator, this test will not catch that. Let's make project do something.

```
.project({"c0 + c1", "c0 * c1"})
```

Reply:
change it.